### PR TITLE
form, validation, item list

### PR DESCRIPTION
**Pull request by MorePositive**
components and logic created

image

### CODE REVIEW

File: src/components/app/app.css [hunk 1]
```diff
@@ -0,0 +1,10 @@
+.App {
+  display: flex;
+  justify-content: center;
+  align-items: center;
+  margin: 0 auto;
+  min-height: 100vh;
+  max-width: 1400px;
+  background-color: #fff;
+  box-shadow: 0px 0px 20px 5px grey;
+}
```
Comment (ShubaraDzmitry) [inline src/components/app/app.css:10]: newline at eof

File: src/components/form/form.js [hunk 2]
```diff
@@ -0,0 +1,160 @@
...
+
+    const { firstName, lastName, phone, email, emailError, emailClazz } = this.state;
+    const { getPost } = this.props;
+
+    return (
+      <form className="form"
+            onSubmit={this.onSubmit}>
+
+        <div className="form-group">
+					<div className="form-wrapper">
```
Comment (ShubaraDzmitry) [inline src/components/form/form.js:89]: what wrong with indents here&? and below

File: src/components/app/app.js [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+
+export default class App extends Component {
+
+  state = {
+    data: []
+  }
+
+  // simple counter to make different dummy requests 
+
+  count = 1;
```
Comment (ShubaraDzmitry) [inline src/components/app/app.js:16]: do we need to use let for this variable?

File: src/components/form/form.js [hunk 4]
```diff
@@ -0,0 +1,160 @@
...
+  // email validation
+
+  validate = () => {
+    const { email } = this.state;
+    if (!email) {
+      this.setState({
+        emailError: "Please enter email",
+        emailClazz: "invalid"
+      })
+      return false
```
Comment (ShubaraDzmitry) [inline src/components/form/form.js:44]: you can track it in state, as you do with a emailError, probably better to rename it to emailError + emailErrorMessage

File: src/components/item-list/item-list.js [hunk 5]
```diff
@@ -0,0 +1,33 @@
+import React, { Component } from 'react';
+
+import './item-list.css';
+
+export default class ItemList extends Component {
```
Comment (ShubaraDzmitry) [inline src/components/item-list/item-list.js:5]: why do we use class component?

File: src/components/item-list/item-list.js [hunk 6]
```diff
@@ -0,0 +1,33 @@
...
+      <div className="item-list">
+        { 
+          !data.length 
+             ?  <p className="no-data">There is no data yet</p>
+             :  data.map(item => {
+                return (
+                  <div  className="item" 
+                        key={item.email || item.id}>
+
+                    <p>{item.firstName || item.userId}</p>
```
Comment (ShubaraDzmitry) [inline src/components/item-list/item-list.js:21]: can we use here ul>li. Looks like more semantic way.

File: src/components/app/app.js [hunk 7]
```diff
@@ -0,0 +1,75 @@
...
+
+  // request to server 
+
+  getPost = (e) => {
+    e.preventDefault();
+    fetch(`https://jsonplaceholder.typicode.com/posts/${this.count++}`)
```
Comment (ShubaraDzmitry) [inline src/components/app/app.js:22]: what if smth went wrong and we don't get response from fetch

**Review by ShubaraDzmitry — CHANGES_REQUESTED** [review]
Good in general, some minor issues

File: src/components/app/app.css (hunk 1 above)
Comment (MorePositive) [reply src/components/app/app.css:10]: Ok

File: src/components/app/app.js (hunk 7 above)
Comment (MorePositive) [reply src/components/app/app.js:22]: Just forgot a catch block. It was my carelessness.
I Will fix it.

File: src/components/form/form.js (hunk 4 above)
Comment (MorePositive) [reply src/components/form/form.js:44]: Ok. Got it.

File: src/components/form/form.js (hunk 2 above)
Comment (MorePositive) [reply src/components/form/form.js:89]: Tryed to make my code more readable. But probably it is a mistake

File: src/components/item-list/item-list.js (hunk 5 above)
Comment (MorePositive) [reply src/components/item-list/item-list.js:5]: I agree with you. This component has been already refactored

File: src/components/item-list/item-list.js (hunk 6 above)
Comment (MorePositive) [reply src/components/item-list/item-list.js:21]: Yes, I didn't attach importance. But of course ul>li is more semantic here.

File: src/components/app/app.js (hunk 3 above)
Comment (MorePositive) [reply src/components/app/app.js:16]: When we initialize a variable inside ES6 classes, we don't need to use let is this case. It will throw an error.

**Pull request merged**